source/server/hot_restarting_parent.cc

```diff
@@ -106,9 +106,15 @@ HotRestartingParent::Internal::getListenSocketsForChild(const HotRestartMessage:
   wrapped_reply.mutable_reply()->mutable_pass_listen_socket()->set_fd(-1);
   Network::Address::InstanceConstSharedPtr addr =
       Network::Utility::resolveUrl(request.pass_listen_socket().address());
+
   for (const auto& listener : server_->listenerManager().listeners()) {
     Network::ListenSocketFactory& socket_factory = listener.get().listenSocketFactory();
-    if (*socket_factory.localAddress() == *addr && listener.get().bindToPort()) {
+    if (*socket_factory.localAddress() == *addr && listener.get().bindToPort() &&
+        (socket_factory.localAddress()->type() != Network::Address::Type::Ip ||
```

Comment:
Why do we check for `socket_factory.localAddress()->type() != Network::Address::Type::Ip` here? Isn't just matching on stream or datagram sufficient?

Reply:
I see, but it seems like leaving this out would future proof it? What's the harm in leaving it?

/wait

Reply:
line 119 to line 121 are only valid for Ip type addresses, and it will incorrectly give socket type datagram for unix sockets. 
We can add a helper function that returns socket type from the address url similar to resolveUrl: https://github.com/envoyproxy/envoy/blob/main/source/common/network/utility.cc#L43

That will give us better future proof.